### PR TITLE
Normalize Gmail addresses to prevent multi-account abuse at signup

### LINKED ISSUE #4086: Prevent email-based multi-account abuse at signup (plus-addressing & disposable domains)
## Problem

Analysis of the last 1,000 suspended accounts shows that the vast majority are automated/bot signups abusing two email tricks to create dozens or hundreds of accounts from a single person:

### 1. Gmail plus-addressing abuse (69.3% of suspended accounts)

A small number of base Gmail accounts create many Gumroad accounts using random `+xxxxx` suffixes. Gmail delivers `user+anything@gmail.com` to `user@gmail.com`, so one inbox controls all accounts.

Top offenders:
- `dentistinprestonpr2@gmail.com` → **101 accounts** (using `+ihkgh`, `+itymu`, `+npsuy`, etc.)
- `affordablehauling7@gmail.com` → **91 accounts**
- `thomaslumholtz5@gmail.com` → **37 accounts**
- `digitalmarketingcomminication1@gmail.com` → **24 accounts**

The plus suffixes are random 5-char strings, clearly script-generated.

### 2. Disposable/bulk domain abuse (17% of suspended accounts)

Four domains account for 154 accounts, all with random gibberish usernames:
- `minofisolu.com` — 47 accounts
- `dofeelowuw.com` — 38 accounts
- `azwiodgebi.com` — 36 accounts
- `link.viet-trade.com` — 33 accounts

### Plus-addressing (no blocklist needed)

At signup, if the email contains `+` and is a Gmail address:
1. Strip the `+suffix` to get the base email
2. Check if the base email (or any plus-addressed variant) already exists in the `users` table
3. If a match exists **and** that account is suspended/flagged, block the signup

This is self-maintaining: the users table itself acts as the blocklist. No new tables or manual maintenance. Each suspended account automatically prevents future plus-addressed variants.

Could also normalize Gmail dots (Gmail ignores dots, so `j.o.h.n@gmail.com` = `john@gmail.com`).

### Disposable domains

Integrate an open-source disposable email domain list (e.g. [disposable-email-domains](https://github.com/disposable-email-domains/disposable-email-domains)) and reject signups from known throwaway domains. Can be updated periodically via a scheduled pull.

## Impact

These two changes would have prevented ~86% of the last 1,000 TOS suspensions from ever creating accounts in the first place.

Filed by Steve Kaye (Risk).

### PR DESCRIPTION
**Pull request by gumclaw**
## What

Gmail plus-addressing & dot normalization behind the `:block_gmail_abuse_at_signup` feature flag: At signup, normalizes Gmail addresses by stripping `+suffix` and removing dots, then checks if any variant belongs to a suspended/flagged account. Blocks signup if so. Uses SQL-level normalization for efficient matching without loading records into memory.

## Why

Analysis of the last 1,000 suspended accounts showed ~69% used Gmail plus-addressing to create dozens/hundreds of accounts from a single person. This change would have prevented the majority from ever creating accounts or selling.

Behind a feature flag for safe rollout.

Disposable email domain blocking has been split into a separate PR: #4263

## Test Results

Tests covering normalization logic, edge cases, and feature flag gating. All passing.

---

AI disclosure: Implemented with Claude Opus 4.6. Prompted to implement the Gmail normalization feature described in #4086, write tests, run simplify, and create PR.

Closes #4086

### CODE REVIEW

File: app/models/concerns/user/email_normalization.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      local, domain = email.downcase.split("@", 2)
+      return email.downcase if domain.blank?
+      return email.downcase if GMAIL_DOMAINS.exclude?(domain)
+
+      local = local.split("+", 2).first
+      local = local.delete(".")
+      "#{local}@gmail.com"
+    end
+
+    def abusive_gmail_variant_exists?(email)
```
Comment (ershad) [inline app/models/concerns/user/email_normalization.rb:25]: This would be a major performance bottleneck unless we add something like an indexed, normalized email column/cache to compare against. Will check our options.

cc: @slavingia
